### ISSUE TITLE
UT - Onboarding

### ISSUE THREAD

**Issue by sarahnw** [CDCgov/prime-reportstream#6407]
The following items, generally in order, must be completed to successfully onboard a receiver to ReportStream:

* [ ] Added to organization.yml
* [ ] Added to organizations-local.yml
* [ ] Added to staging settings table
* [ ] Added to prod settings table
* [ ] Added at least one receiver
* [ ] Added OKTA group to download site
* [ ] Added customers to OKTA group for download site
* [ ] Is customer contact made?
* What doc format are we supporting for the receiver (HLV, CSV, other)? **(Add comment below)**
* What transport(s) are we supporting for the receiver (SFTP, other)? **(Add comment below)**
* [ ]  Got credentials?
* [ ] Tested ELR connection locally?
* [ ] Tested ELR connection from Azure?
* [ ] Has the receiver accessed the download site?
* [ ] Has the receiver been sent at least one test file?
* [ ] Has the receiver signed off on file formatting?
* [ ] Are there labs sending data?
* [ ] Has the receiver confirmed receipt of the data?
* [ ] Have the files processed successfully?
* [ ] Are we in production?
* [ ] Update customerStatus flag in production settings to active

**Comment by oslynn** [comment]
2022-09-30.  Sarah sent an email requesting information to connect them up with ReportStream.

**Comment by sliu1000** [comment]
Email sent to Sararh on 10/7/2022:

Sarah

I apologize for the delay in response. Jason went out on paternity leave almost immediately after the call and we haven't had time to debrief Rachelle about what was discussed. I can answer some of these questions but not all of them.
We do have an implimentation guide we can send. It's a little bit out of date but I'm currently working on updating it. If you're willing to wait a week or two I can send that out.
We don't require a specific name format
We do accept v2.5.1, in fact we prefer it!
Our usual protocol for accepting files is SFTP. When we begin onboarding we will request your static IP addresses in order to whitelist them and we will create the SFTP for you. When the account is built and the firewall exception is complete we will send you your credentials, along with the file path you'll need. Our SFTP connections do not have a rotation or expiration.
Our system does accept batched messages and we are not aware of any limit to that. Our system functions better with batched reporting and every 2 hours will work great for us.
We are happy with any identifier in MSH-5 and -6. Our usual identifiers are:
MSH.5 = NEDSS
MSH.6 = usually UDOH but can be any indicator 
We do require an identifier in MSH.4.1. The remainder of the MSH.4 segment is nice to have but not neccesary
We would like to receive the AOE questions
I'm unsure about the QST flag - I'll ask Jason when he returns.
Our onboarding form provides most of the required elements. The official information from UDHHS is located at https://epi.health.utah.gov/disease-reporting/. however 
We don't need you to suppress the QST flag, as far as we know.
We do have a series of fields we strongly recommend but we will make do without them. Unless there is no patient demographic data or no test result we really don't throw away any chance for data. The fields we prefer to be included are in the implementation guide we can send out in about a week.
We don't require a specific value in MSH.21.1
We only want to receive OTC results if they're ready by an app/computer or by qualified personnell (i.e. nurse, lab tech). If they are home performed and read without an app we do not want to receive them. The OBX. field should be listed as "At-Home Test" for any OTC tests performed and read by an app.
I apologize again for the delay in this email and in the start of this process. Since our kickoff meeting we've had quite a few changes within our electronic data exchange team. I have accepted the position of EDX team lead and can be your point of contact. If you email here and don't get a response feel free to reach out directly to my email at teribills@utah.gov

Let us know what our next steps are so we can get things started!

Teri

**Comment by oslynn** [comment]
2022-10-14: Emailed Teri for SFTP credential and gave IP for whitelist.

"===
Good After noon, Teri,

My Name is OTT SATHNGAM.  I am responsible for onboarding your organization with ReportStream.  I completed ReportStream configuration setting for your organization.
I ready to send you the sample hl7 file for validation.  Please provide me your SFTP information such as credential and path to your staging and production either by KeyBase
(my KeyBase Id: oslynn1) or by secured email.  Below is our ReportStream's IP address and my test system for white listing.

ReportStream IPs:  158.111.236.0/24 (i.e 158.111.236.0, 158.111.236.1, 158.111.236.2, … , 158.111.236.24)
Test system IP:  45.26.16.41

Please notify me when I can start SFTP the sample hl7 for evaluation.

Very best regards,
-Ott Sathngam
CDC SWE

**Comment by brick-green** [comment]
Any communication from UT on the whitelisting?

**Comment by sliu1000** [comment]
@brick-green no we have not. I can follow-up with them.

**Comment by oslynn** [comment]
2022-10-31: Followed up with Teri Bill.

Teri,

How are you?  Would you please give me the status of your Setting Up your SFTP site?  Do you have the IP for us to whitelist it?

Very best regards,
-Ott Sathngam
CDC SWE

**Comment by BerniXiongA6** [comment]
11/4/2022: BX email to Rachelle Bouton:

> Hi Rachelle,
> 
> I received an auto-reply email that you were the correct person to contact while Teri is out of the office. Would you happen to have a status update on Utah's SFTP connection set-up progress?
> 
> 
> Thanks,
> Berni

11/4/2022: Teri auto-reply message to BX:

> I will be out of the office until November 14, 2022. I will respond as quickly as I am able when I return.  For general questions please contact edx@utah.gov. For any emergency questions please contact Rachelle Boulton at rboulton@utah.gov.
> 
> Thank you!
> 
> Teri Bills

11/4/2022: BX email to Teri Bills:

> Hi Teri,

> 
> I wanted to quickly make a few introductions in this follow-up email to you post-kickoff call with the CDC PRIME project members, since we've not yet had the chance to meet. My name is Berni Xiong, and I'm the new delivery manager on ReportStream's Onboarding and Operations teams. I work closely with Brandon Canniff (copied in this email as well) who has recently joined our team (replacing Sarah White). Together, with the support of our amazing engineering team, we'll be able to assist you with any questions you have during your onboarding experience with ReportStream.
> 
> In reviewing the previous email threads, it sounds as though we're waiting to get the green light from your team that you've set up the SFTP connection, is that correct? 
> 
> Our engineers are ready to assist when your team is, so please let us know if there's anything we can do to help. We look forwarding to continuing on with the onboarding process with your team!
> 
> Thanks,
> Berni
>

**Comment by BerniXiongA6** [comment]
11/4/2022: Waiting on Rachelle Boulton to hopefully provide a status update in Teri Bills' absence.

**Comment by BerniXiongA6** [comment]
BX sent email to Teri and Rachelle on 11/21:

> Hi Teri and Rachelle,
> 
> Brick is currently out on PTO this week, so I wanted to make sure I reached out in the event either of you had emailed him to follow up on continuing with the onboarding process.
> 
> I realize that it's a holiday week; so if you decide it's better to pick back up where we left off either next week or December, our team is happy to assist. Let us know what works best for you.
> 
> Thanks,
> Berni

**Comment by oslynn** [comment]
I just followed up with them.  And, I still haven't receive any answer, yet.

**Comment by brandonnava** [comment]
@oslynn requested a firewall to talk to their system. No response from CDC yet. Requested @jeremy-page for access

**Comment by sliu1000** [comment]
Update from UTAH:

Hello,

We are set up to receive through the sFTP. Please us know when file has been sent over.

Thank you,

Rocio
On Fri, Jan 13, 2023 at 1:34 PM Sathngam, Ott (CDC/DDPHSS/OD/HITSSU) <sry8@cdc.gov> wrote:
Good after noon, Rocio,

Is UT SFTP ready to accept production test result?  If yes, I will activate ReportStream to route test result to your organization, this Monday.

Very respectfully,
-Ott Sathngam
CDC SWE

**Comment by sliu1000** [comment]
Emailed UTAH and advised them that they are LIVE.

From: Report Stream (CDC) <reportstream@cdc.gov>
Sent: Thursday, January 19, 2023 3:33 PM
To: Utah DCP Informatics Program <edx@utah.gov>; Rocio Ramos (DHHS) <rramos@utah.gov>
Cc: Sathngam, Ott (CDC/DDPHSS/OD/HITSSU) <sry8@cdc.gov>; Reeves, Maurice (CDC/OD/OCS) <qva8@cdc.gov>; Green, Brick (CDC/DDPHSS/OD/HITSSU) (CTR) <rut5@cdc.gov>; james.gilmore@agile6.com <james.gilmore@agile6.com>; brandoncanniff@navapbc.com <brandoncanniff@navapbc.com>
Subject: Re: CDC’s ReportStream Kickoff - Feedback Requested
 
Hi Rocio,

I am happy to inform that you are now connected to ReportStream Production.  Please be sure to monitor your feed and let us know if you need anything further.

Best regards,

Sharon
The ReportStream Support Team

Contact us at reportstream@cdc.gov

Support hours:  8:00 am through 9:00 pm Eastern time

Monday-Friday, excluding US public holidays

From: rramos@utah.gov <rramos@utah.gov> on behalf of Utah DCP Informatics Program <edx@utah.gov>
Sent: Wednesday, January 18, 2023 1:39 PM
To: Report Stream (CDC) <reportstream@cdc.gov>
Cc: Sathngam, Ott (CDC/DDPHSS/OD/HITSSU) <sry8@cdc.gov>; Xiong, Bernadette (CDC/DDPHSS/OD/HITSSU) (CTR) <rwe7@cdc.gov>; Reeves, Maurice (CDC/OD/OCS) <qva8@cdc.gov>; Green, Brick (CDC/DDPHSS/OD/HITSSU) (CTR) <rut5@cdc.gov>; james.gilmore@agile6.com <james.gilmore@agile6.com>; brandoncanniff@navapbc.com <brandoncanniff@navapbc.com>; Rocio Ramos (DHHS) <rramos@utah.gov>
Subject: Re: CDC’s ReportStream Kickoff - Feedback Requested
 
Perfect. Thank you.
On Wed, Jan 18, 2023 at 11:00 AM Report Stream (CDC) <reportstream@cdc.gov> wrote:
Hi Rocio,

Thank you for your email.  I am confirming receipt.  Will reach out to the Engineer working on this and they will get back to you shortly.

Best regards,

Sharon
The ReportStream Support Team

Contact us at reportstream@cdc.gov

Support hours:  8:00 am through 9:00 pm Eastern time

Monday-Friday, excluding US public holidays

From: rramos@utah.gov <rramos@utah.gov> on behalf of Utah DCP Informatics Program <edx@utah.gov>
Sent: Wednesday, January 18, 2023 12:14 PM
To: Sathngam, Ott (CDC/DDPHSS/OD/HITSSU) <sry8@cdc.gov>
Cc: Xiong, Bernadette (CDC/DDPHSS/OD/HITSSU) (CTR) <rwe7@cdc.gov>; Report Stream (CDC) <reportstream@cdc.gov>; Reeves, Maurice (CDC/OD/OCS) <qva8@cdc.gov>; Green, Brick (CDC/DDPHSS/OD/HITSSU) (CTR) <rut5@cdc.gov>; james.gilmore@agile6.com <james.gilmore@agile6.com>; brandoncanniff@navapbc.com <brandoncanniff@navapbc.com>; Rocio Ramos (DHHS) <rramos@utah.gov>
Subject: Re: CDC’s ReportStream Kickoff - Feedback Requested
 
Hello,

We are set up to receive through the sFTP. Please us know when file has been sent over.

Thank you,

Rocio

**Comment by oslynn** [comment]
Done.
